clash-prelude/src/Clash/Explicit/Reset.hs

```diff
@@ -206,31 +283,43 @@ resetGlitchFilter
   -- reset asserted for /glitchlessPeriod/ cycles straight.
   -> Clock dom
   -> Reset dom
+  -- ^ The power-on reset for the glitch filter itself
   -> Reset dom
-resetGlitchFilter SNat clk rst =
-  case initBehavior @dom of
-    SDefined ->
-      unsafeToReset $
-        moore clk noReset enableGen go fst (asserted, 0) (unsafeFromReset rst)
-    SUnknown ->
-      error "'resetGlitchFilter' only supports domains with defined initial values."
+  -- ^ The reset that will be filtered
+  -> Reset dom
+resetGlitchFilterWithReset glitchlessPeriod clk ownRst =
+  resetGlitchFilter# glitchlessPeriod (register clk ownRst enableGen)
+
+resetGlitchFilter#
+  :: forall dom glitchlessPeriod n state
+   . ( HasCallStack
+     , KnownDomain dom
+     , glitchlessPeriod ~ (n + 1)
+     , state ~ (Bool, Index glitchlessPeriod)
+     )
+  => SNat glitchlessPeriod
+  -> (   state
+      -> Signal dom state
+      -> Signal dom state
+     )
+  -> Reset dom
+  -> Reset dom
+resetGlitchFilter# SNat reg rstIn =
+  let s' = go <$> s <*> unsafeFromReset rstIn
+      s  = reg (asserted, 0) s'
+  in unsafeToReset $ fst <$> s
  where
-  go :: state ~ (Bool, Index glitchlessPeriod) => state -> Bool -> state
+  go :: state -> Bool -> state
   go (state, count) reset
     | reset == state    = (state,     0)
     | count == maxBound = (not state, 0)
     | otherwise         = (state,     count + 1)
 
-  noReset :: Reset dom
-  noReset = unsafeToReset (pure (not asserted))
-
   asserted :: Bool
   asserted =
     case resetPolarity @dom of
       SActiveHigh -> True
       SActiveLow -> False
--- See: https://github.com/clash-lang/clash-compiler/pull/2511
-{-# CLASH_OPAQUE resetGlitchFilter #-} -- Give reset glitch filter its own HDL file
```

Comment:
Why don't we want this any longer?

Reply:
I asked Martijn why he did this and he said it was just for nicer signal/port names for the filter. You expressed your disapproval about that a while back.

But it should have been mentioned in the commit message, I'll add it.